File: src/main/java/me/redepicness/bingo/world/BingoWorldsConfig.java
package me.redepicness.bingo.world;

import org.jetbrains.annotations.NotNull;

public record BingoWorldsConfig(@NotNull BingoWorldPreloadConfig preloading) {

    public BingoWorldsConfig(){
        this(new BingoWorldPreloadConfig());
    }

    public record BingoWorldPreloadConfig(boolean preload, int overworldRadius, int netherRadius, int endRadius){

        public BingoWorldPreloadConfig(){
            this(true, 5000, 1000, 200);
        }

    }

}

File: src/main/kotlin/me/redepicness/bingo/world/BingoWorldsConfig.kt
package me.redepicness.bingo.world

data class BingoWorldsConfig(
        val preloading: BingoWorldPreloadConfig = BingoWorldPreloadConfig(),
)

data class BingoWorldPreloadConfig(
        val preload: Boolean = true,
        val overworldRadius: Int = 5000,
        val netherRadius: Int = 1000,
        val endRadius: Int = 200,
)
